libs/langchain/tests/unit_tests/tools/test_convert_to_openai.py

```diff
@@ -0,0 +1,60 @@
+from langchain_community.tools.google_serper.tool import GoogleSerperRun
+from langchain_community.utilities.google_serper import GoogleSerperAPIWrapper
+from langchain_core.tools import tool, Tool
+from langchain.tools import format_tool_to_openai_function
+from langchain.chains import LLMMathChain
+from langchain.llms import OpenAI
+import os
+
+
+def assert_oai_func(oai_func_desc, parameter_name):
+    assert 'parameters' in oai_func_desc
+    assert 'properties' in oai_func_desc['parameters']
+    parameter = oai_func_desc['parameters']['properties']
+    assert len(parameter) == 1
+    assert parameter_name in parameter
+
+
+def test_function_tool() -> None:
+    @tool
+    def get_word_length(word: str) -> int:
+        """Returns the length of a word."""
+        return len(word)
+
+    oai_func_desc = format_tool_to_openai_function(get_word_length)
+    assert_oai_func(oai_func_desc, 'word')
+
+
+def test_subclass_of_base_tool() -> None:
+    """test tools created by subclassing BaseTool"""
+    os.environ['SERPER_API_KEY'] = '1' * 40  # mocked key
+    serp_tool = GoogleSerperRun(api_wrapper=GoogleSerperAPIWrapper())
+    oai_func_desc = format_tool_to_openai_function(serp_tool)
+    assert_oai_func(oai_func_desc, 'query')
+
+
+def test_class_of_tool() -> None:
+    def get_word_length(word: str) -> int:
+        """Returns the length of a word."""
+        return len(word)
+
+    t = Tool(
+        name="word_length",
+        description="Useful for count word length.",
+        func=get_word_length,
+    )
+    oai_func_desc = format_tool_to_openai_function(t)
+    assert_oai_func(oai_func_desc, 'word')
+
+
+def test_class_of_tool_with_chain_run() -> None:
+    os.environ['OPENAI_API_KEY'] = 'sk-' + '1' * 49
```

Comment:
this seems potentially bad - can we use a FakeLLM

Reply:
Thanks for pointing it out. Use FakeLLM instead.